extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/DefaultTenantConfigResolver.java

```diff
@@ -152,30 +152,48 @@ private Uni<TenantConfigContext> initializeTenantIfContextNotReady(TenantConfigC
     }
 
     private TenantConfigContext getStaticTenantContext(RoutingContext context) {
-
         String tenantId = context.get(CURRENT_STATIC_TENANT_ID);
-
         if (tenantId == null && context.get(CURRENT_STATIC_TENANT_ID_NULL) == null) {
-            if (tenantResolver.isResolvable()) {
-                tenantId = tenantResolver.get().resolve(context);
+            tenantId = resolveStaticTenantId(context);
+            if (tenantId != null) {
+                context.put(CURRENT_STATIC_TENANT_ID, tenantId);
+            } else {
+                context.put(CURRENT_STATIC_TENANT_ID_NULL, true);
             }
+        }
+
+        return getStaticTenantContext(tenantId);
+    }
+
+    private String resolveStaticTenantId(RoutingContext context) {
+        String tenantId;
+        if (tenantResolver.isResolvable()) {
+            tenantId = tenantResolver.get().resolve(context);
 
-            if (tenantId == null && tenantConfigBean.getStaticTenantsConfig().size() > 0) {
-                tenantId = defaultStaticTenantResolver.resolve(context);
+            if (tenantId != null) {
+                return tenantId;
             }
+        }
 
-            if (tenantId == null) {
-                tenantId = context.get(OidcUtils.TENANT_ID_ATTRIBUTE);
+        if (pathMatchingTenantResolver != null) {
+            tenantId = pathMatchingTenantResolver.resolve(context);
+
+            if (tenantId != null) {
+                return tenantId;
             }
         }
 
+        tenantId = context.get(OidcUtils.TENANT_ID_ATTRIBUTE);
```

Comment:
I think this attribute check should be above the path based check. If it is set (with `TenantId`, or, in case of the code flow, when the session cookie has it its name) then all the automatic checks (path matching and convention based) can be skipped. Only the custom `TenantResolver` where the custom code can make some decisions even if this attribute is set, should run first.

Reply:
If agreed, please rearrange the section is the docs

Reply:
I'll follow up later with a clarification re the code flow tenant id resolution from the session cookie

Reply:
Sure. In additional, I find it more natural that annotation-based selection has higher priority because it is more specific in my personal view.